### PR TITLE
Add DEP template, restructure docs

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
Add DEP template, restructure multi-node etcd clusters proposal to DEP-01. Also restructure other files in `docs` folder:
- Consolidated `compaction-metrics.md` and `multinode-metrics.md` into a single `docs/operations/metrics.md` doc
- Replace `_` with `-` in doc filenames
- Restructure `docs/proposals` section:
  - Add DEP (Druid Enhancement Proposal) template, for proposals of large design changes in etcd-druid
  - Multi-node etcd proposal becomes DEP-01
  - Snapshot compaction becomes DEP-02
  - Scale-up of single-node to multi-node etcd cluster becomes DEP-03
- Replace occurrences of `ETCD` with `Etcd` or `etcd`, since ETCD is not an abbreviation, and should not be used in all-caps

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc developer
Introduce DEPs (Druid Enhancement Proposals) for proposing large design changes in etcd-druid.
```

### CODE REVIEW

File: docs/proposals/00-template.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+status: implementable|implemented
+authors:
+- "@you"
+- "@maybe-another-author"
+reviewers:
+- "@main-reviewer-1"
+- "@main-reviewer-2"
+---
+
+# GEP-NN: Your short, descriptive title
```
Comment (shafeeqes) [inline docs/proposals/00-template.md:14]: DEP

Comment (shreyas-s-rao) [reply docs/proposals/00-template.md:14]: Nice catch, thanks

**Comment by shreyas-s-rao** [comment]
@shafeeqes addressed, PTAL

**Review by aaronfern — APPROVED** [review]
Thanks for the PR!
One minor nit, else all good

/lgtm

**Comment by shreyas-s-rao** [comment]
@aaronfern addressed, along with few more replacements of `ETCD` with `etcd`, since `ETCD` is technically the incorrect term to use since it isn't an abbreviation. PTAL

**Comment by ishan16696** [comment]
> Scale-up of single-node to multi-node etcd cluster becomes DEP-03

Scale-up has already been implemented and it's a feature now. 
I don't get it why you have put it in `proposals` ? It's no longer proposal anymore.

**Comment by shreyas-s-rao** [comment]
@ishan16696 It's a design doc, and design is usually documented in proposals, since the general flow is to create a proposal with design changes and then implement it. Additionally, since the scale-up logic was changed from the original multi-node proposal, it should either be added as an amendment/edit to the multi-node proposal itself, or as a new proposal that overwrites certain sections of the original multi-node proposal. Hence, I added it to the `Proposals` section.

**Pull request merged**